contracts/PortfolioLib.sol

```diff
@@ -249,6 +249,9 @@ function getPoolReserves(PortfolioPool memory self)
     pure
     returns (uint128 reserveAsset, uint128 reserveQuote)
 {
+    // Check if -`self.liquidity` fits within an int128 by checking if it's
+    // greater than the minimum negative value of an int128.
+    if (self.liquidity > 170141183460469231731687303715884105728) revert();
```

Comment:
I'd say writing it as `if (self.liquidity > 2 ** 127)` is more obvious than this base10 decimal number and the compiler will replace the exponentiation at compile time, so there's no gas overhead.

Reply:
Makes sense, fixed now!

Reply:
fixed